great_expectations/datasource/data_connector/util.py

```diff
@@ -290,6 +288,38 @@ def get_filesystem_one_level_directory_glob_path_list(
     return path_list
 
 
+def list_azure_keys(
+    azure: BlobServiceClient,
+    query_options: dict,
+    container: str,
+    iterator_dict: dict,
+    recursive: bool = False,
+) -> List[str]:
```

Comment:
`list_s3_keys` uses a generator to yield `str` but do we need to adhere to that? I think we might be able to just return a list here?